deploy/gcp/main.tf

```diff
@@ -127,15 +139,16 @@ resource "google_container_node_pool" "pd_pool" {
 }
 
 resource "google_container_node_pool" "tikv_pool" {
-  provider           = "google-beta"
-  project            = "${var.GCP_PROJECT}"
-  cluster            = "${google_container_cluster.cluster.name}"
-  location           = "${google_container_cluster.cluster.location}"
+  depends_on         = [google_container_node_pool.pd_pool]
```

Comment:
Why do we need these new dependencies for node pool?

Reply:
GCP locks node pool resources so you can't delete them concurrently. So when doing `terraform destroy` it would abort with an error. This makes it so the user doesn't have to re-enter the command. I couldn't find a different way to enforce order only when doing `terraform destroy` and not during `terraform apply`